src/AnnotatedCommand.php

```diff
@@ -247,12 +248,22 @@ private function addInputOption($inputOption, $description = null)
             $default = null;
         }
 
+        $suggestedValues = [];
```

Comment:
Rather than doing this, could we call `$definition = $this->getDefinition()` and then `$definition->addOption($inputOption)`? We might need to do the same with `getNativeDefinition()`, and implementation might need to vary with Symfony version. See [Commmand.php](https://github.com/symfony/console/blob/a575bcb51293996ad54127bd43eb84c83013cdc5/Command/Command.php#L416,L479)

Reply:
I think there's a reason we didn't do this in older versions of Symfony, but it looks like ^^ would work in ^6.1

Reply:
I'd rather avoid breaking encapsulation if we can manage it.

Reply:
I pushed a commit containing what I think you are suggesting. Its cleaner, but there is a problem. As shown below, we have lost the description for the --format option. You can see in the code that we doing nothing with that $description parameter now. The InputOption class provides no way to set a description except in its constructor. So I think we are right back where we started, having to get the suggestedValues out of an existing InputOption instance and build a new InputOption with right description.

```
weitzman@drush-web:/var/www/html$ drush st -h
An overview of the environment - Drush and Drupal.

Examples:
 drush core-status --field=files Emit just one field, not all the default fields.
 drush core-status --fields=*    Emit all fields, not just the default ones.

Arguments:
 [filter] A field to filter on. @deprecated - use --field option instead.

Options:
 --project=PROJECT A comma delimited list of projects. Their paths will be added to path-aliases section.
 --format[=FORMAT]  [default: table]
 --fields=FIELDS   Available fields: Drupal version (drupal-version), Site URI (uri), DB driver (db-driver), DB hostname (db-hostname), DB port
                   (db-port), DB username (db-username), DB password (db-password), DB name (db-name), Database (db-status), Drupal bootstrap
                   (bootstrap), Default theme (theme), Admin theme (admin-theme), PHP binary (php-bin), PHP config (php-conf), PHP OS (php-os), PHP
                   version (php-version), Drush script (drush-script), Drush version (drush-version), Drush temp (drush-temp), Drush configs
                   (drush-conf), Drush aliases (drush-alias-files), Alias search paths (alias-searchpaths), Install profile (install-profile),
                   Drupal root (root), Drupal Settings (drupal-settings-file), Site path (site-path), Site path (site), Themes path (themes),
                   Modules path (modules), Files, Public (files), Files, Private (private), Files, Temp (temp), Drupal config (config-sync), Files,
                   Public (files-path), Files, Temp (temp-path), Other paths (%paths) [default:
                   drupal-version,uri,db-driver,db-hostname,db-port,db-username,db-name,db-status,bootstrap,theme,admin-theme,php-bin,php-conf,php-o
                   s,php-version,drush-script,drush-version,drush-temp,drush-conf,install-profile,root,site,files,private,temp]
 --field=FIELD     Select just one field, and force format to *string*.
```